### ISSUE TITLE
Failed to register CUDA shared memory region 'fc6_1'

### ISSUE THREAD

**Issue by santapo** [triton-inference-server/server#4479]
**Description**
When running `perf_analyzer` with `shared-memory=cuda` for the `densenet_onnx` from docs example, I've met this error
```
WARNING: CUDA Minor Version Compatibility mode ENABLED.
  Using driver version 495.29.05 which has support for CUDA 11.5.  This container
  was built with CUDA 11.7 and will be run in Minor Version Compatibility mode.
  CUDA Forward Compatibility is preferred over Minor Version Compatibility for use
  with this container but was unavailable:
  [[System has unsupported display driver / cuda driver combination (CUDA_ERROR_SYSTEM_DRIVER_MISMATCH) cuInit()=803]]
  See https://docs.nvidia.com/deploy/cuda-compatibility/ for details.

error: failed to create concurrency manager: failed to register CUDA shared memory region 'fc6_1'
```

**Triton Information**
I used Triton Server dockerfile: `nvcr.io/nvidia/tritonserver:20.05-py3` and `tritonclient==2.22.0`  from pip

**To Reproduce**
1. Run Triton Server with example model repository that contains `densenet_onnx`
```bash
#!/bin/sh
export CUDA_VISIBLE_DEVICES=0
docker run --gpus=0 --rm -p 8080:8080 -p 8081:8081 -p 8082:8082 --net=host \
            -v full-path-to-triton-inference-server/model_repository:/models nvcr.io/nvidia/tritonserver:22.05-py3 tritonserver \
            --model-repository=/model
```
- DenseNet Onnx configuration
```
name: "densenet_onnx"
platform: "onnxruntime_onnx"
max_batch_size : 0
input [
  {
    name: "data_0"
    data_type: TYPE_FP32
    format: FORMAT_NCHW
    dims: [ 3, 224, 224 ]
    reshape { shape: [ 1, 3, 224, 224 ] }
  }
]
output [
  {
    name: "fc6_1"
    data_type: TYPE_FP32
    dims: [ 1000 ]
    reshape { shape: [ 1, 1000, 1, 1 ] }
    label_filename: "densenet_labels.txt"
  }
]
```
2. Run Triton Client
```bash
#!/bin/sh
export CUDA_VISIBLE_DEVICES=0
docker run --gpus=all --net=host nvcr.io/nvidia/tritonserver:22.05-py3-sdk perf_analyzer \
        -m densenet_onnx --percentile=95 --shared-memory=cuda
```

**Expected behavior**
Get evaluated results when using `shared-memory=cuda`

**Comment by tanmayv25** [comment]
Unfortunately complete error message is not being reported. With [this](https://github.com/triton-inference-server/server/pull/4417) fix we should now report the complete error message. This will include the reason for failure. We have the feature being successfully tested in our CI [here](https://github.com/triton-inference-server/server/blob/main/qa/L0_perf_analyzer/test.sh#L205), so most likely the issue is related to your specific setup.

@santapo Can you make sure that you don't have any pre-allocated cuda shared memory region registered on the machine? By any chance do you have any other instance of trionserver running?
Also I find the below warning concerning. Can you try upgrading your driver?
```
[[System has unsupported display driver / cuda driver combination (CUDA_ERROR_SYSTEM_DRIVER_MISMATCH) cuInit()=803]]
  See https://docs.nvidia.com/deploy/cuda-compatibility/ for details.
 ```

**Comment by santapo** [comment]
I've checked cuda allocated shared memory by using `cudashm` of `tritonclient` like code block below and it returns an empty list. Could `cudashm.allocated_shared_memory_regions()` detect pre-allocated shared memory regions or it only detect regions that registered by `tritonclient`?
```python
>>> import tritonclient.utils.cuda_shared_memory as cudashm                                                                                                                                                       
>>> cudashm.allocated_shared_memory_regions()                                                                                                                                                                     
[]   
```

**Comment by tanmayv25** [comment]
@santapo  You are running perf_anlalyzer in a separate container than Triton's. In order to use system and cuda shared memory across the containers, we recommend launching the triton container like:

`docker run --gpus=0 -it --rm --init --shm-size=10G --ulimit memlock=-1 --net=host  ... `

**Comment by Tabrizian** [comment]
Adding to @tanmayv25's comment you need to add `--ipc host` to both the SDK and Server container.

**Comment by sunkenQ** [comment]
I ran into the same problem, when Client docker on machine B called Server Docker on machine A to execute Perf Analyzer, I got **"[failed to register CUDA shared memory region 'output_layer](error: failed to create concurrency manager: failed to register CUDA shared memory region 'output_layer')'"**, but simple_grpc_cudashm_client.py worked fine. I also use "cudashm.allocated_shared_memory_regions()" check pre-allocated cuda and return value is empty list.
What do I need to try if I want to execute Perf Analyzer in Machine B?

Machine A
  docker: nvcr.io/nvidia/tritonserver:21.03-py3  
  nvidia driver: 460.73.01
  docker run cmd:  docker run  -it --gpus 0 --shm-size=10gb --rm --init --ulimit memlock=-1 --ipc=host --network=host -p 8000:8000 -p 8001:8001 -p 8002:8002 -v /model_repository:/models nvcr.io/nvidia/tritonserver:21.03-py3 tritonserver --model-repository=/models  --grpc-infer-allocation-pool-size=16
Machine B 
  docker nvcr.io/nvidia/tritonserver:21.03-py3-sdk
  nvidia driver: 470.82.01
  docker run cmd: docker run --ipc host -it --gpus 0 -v /var/run/docker.sock:/var/run/docker.sock --net=host nvcr.io/nvidia/tritonserver:21.03-py3-sdk
